templates/BuildMyCharm/config.xml

```diff
@@ -84,7 +103,7 @@ fi
     <org.jenkinsci.plugins.postbuildscript.PostBuildScript plugin="postbuildscript@0.17">
       <buildSteps>
         <hudson.tasks.Shell>
-          <command>juju destroy-model test-{{charmname}}-$BUILD_NUMBER -y</command>
+          <command>juju destroy-model test-{{charmname}}-$BUILD_NUMBER -y || true</command>
```

Comment:
You're right.  I switched it to use `trap cleanup_model EXIT` instead of post-build which should preserve errors and still be clean.  We should really try to figure out why it sometimes runs the job immediately after submit and sometimes doesn't; I guess it has to do with timing of when the job is submitted.